src/components/ConnectorToolbar.tsx

```diff
@@ -135,35 +135,29 @@ function ConnectorToolbar({
     ]);
 
     const handlers = {
-        setProtocol: debounce(
-            (_event: SyntheticEvent, value: string | null) => {
-                const selectedProtocol = protocolOptions.find(
-                    (option) => option.message === value
-                )?.protocol;
+        setProtocol: (_event: SyntheticEvent, value: string | null) => {
+            const selectedProtocol = protocolOptions.find(
+                (option) => option.message === value
+            )?.protocol;
 
-                setProtocol(selectedProtocol ? selectedProtocol : null);
-            },
-            750
-        ),
-        setSortBy: debounce((_event: SyntheticEvent, value: string | null) => {
+            setProtocol(selectedProtocol ? selectedProtocol : null);
+        },
+        setSortBy: (_event: SyntheticEvent, value: string | null) => {
```

Comment:
No need to debounce these as users are not directly typing in values. Just made things feel slow.